src/frontend/src/optimizer/mod.rs

```diff
@@ -250,6 +235,19 @@ impl PlanRoot {
             ctx.trace(plan.explain_to_string());
         }
 
+        // Convert to physical plan node
+        plan = plan.to_batch_with_order_required(&self.required_order)?;
+        if ctx.is_explain_trace() {
+            ctx.trace("To Batch Plan:");
+            ctx.trace(plan.explain_to_string());
+        }
+
+        plan = plan.optimize_by_rules(&OptimizationStage::new(
+            "Merge BatchProject",
+            vec![BatchProjectMergeRule::create()],
+            ApplyOrder::BottomUp,
+        ));
```

Comment:
Shall we do this before "Convert to physical plan node"?

Reply:
This rule is necessary, because `Convert to physical plan node` might generate additional project.